spec/models/mixins/supports_feature_mixin_spec.rb

```diff
@@ -311,17 +311,19 @@ def initialize(values = {})
 
   describe ".providers_supporting" do
     it 'detect' do
-      providera = define_subclass("ProviderA", ExtManagementSystem)
-      providerb = define_subclass("ProviderB", ExtManagementSystem)
-      providerc = define_subclass("ProviderC", ExtManagementSystem)
+      stub_const("ExtManagementSystem", model)
```

Comment:
This is the main change here.

When we call `providers_supporting` / and therefore `Ems.subclasses` - it will fetch the subclasses from our generated/stubbed tree. So we've successfully created a separte tree and no longer need to modify the real Ems tree.

Reply:
Do we want to add a comment like "`ExtManagementSystem` is used by `providers_supporting`"?

Reply:
Yeah I could